### PR TITLE
Add cleanup-on-success option to repackager

### PR DESCRIPTION
**Pull request by jeremyh**
- Adds a `--clean-inputs` option which removes the original input dataset after successful recompress. (needed to avoid doubling the disk usage)
- Improves logging. Switched it to structlog.
- Add a `-f` option to read a list of inputs from a file (one per line)
- Various test improvements for edge cases

### CODE REVIEW

File: eodatasets/scripts/recompress.py [hunk 1]
```diff
@@ -371,20 +404,43 @@ def main(paths: List[str], output_base: str, zlevel: int, block_size: int):
...
+        _LOG.info(
+            "node.finish",
+            host=socket.getfqdn(),
+            total_count=total,
+            failure_count=failures,
+        )
     sys.exit(failures)
 
 
+def please_remove(path: Path, excluding: Path):
```
Comment (sixy6e) [inline eodatasets/scripts/recompress.py:428]: so polite !!! :)

**Review by sixy6e — APPROVED** [review]
All looks good to me!
